### PR TITLE
Make sizeof() work for external symbols

### LINKED ISSUE #1728: sizeof(x) doesn't work for external symbol `x`
```
semantic error: Symbol 5 is not yet supported in sizeof.
   --> a.py:313:21
    |
313 |             ctx_id, sizeof(common_dev_host),
    |                     ^^^^^^^^^^^^^^^^^^^^^^^ 
```

### PR DESCRIPTION
**Pull request by certik**
Fixes #1728.

### CODE REVIEW

**Comment by certik** [comment]
Right now I am getting:
```
ASR verify pass error: ASR verify: Struct::m_derived_type cannot point outside of its symbol table
```
So more fixes are needed.

**Review by czgdp1807 — COMMENTED** [review]
Could you please add a test? I would like to see the stacktrace to figure out whether this fix is correct.

File: src/lpython/semantics/python_ast_to_asr.cpp [hunk 1]
```diff
@@ -6567,13 +6567,14 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
...
-                    if( ASR::is_a<ASR::Variable_t>(*arg_Var->m_v) ) {
-                        arg_type = ASR::down_cast<ASR::Variable_t>(arg_Var->m_v)->m_type;
-                    } else if( ASR::is_a<ASR::StructType_t>(*arg_Var->m_v) ) {
+                    ASR::symbol_t *arg_Var2 = ASRUtils::symbol_get_past_external(arg_Var->m_v);
+                    if( ASR::is_a<ASR::Variable_t>(*arg_Var2) ) {
+                        arg_type = ASR::down_cast<ASR::Variable_t>(arg_Var2)->m_type;
+                    } else if( ASR::is_a<ASR::StructType_t>(*arg_Var2) ) {
                         arg_type = ASRUtils::TYPE(ASR::make_Struct_t(al, x.base.base.loc,
-                                        arg_Var->m_v, nullptr, 0));
+                                        arg_Var2, nullptr, 0));
```
Comment (czgdp1807) [inline src/lpython/semantics/python_ast_to_asr.cpp:6575]: Actually here you should use `arg_Var->m_v` (i.e., the original external symbol). I am making a new PR with a test.

**Comment by czgdp1807** [comment]
Closing in favour of https://github.com/lcompilers/lpython/pull/1730

**Pull request closed without merging**